observer/observer.go

```diff
@@ -202,6 +207,38 @@ func (o *Observer) IsAnyJobRunning(repository string) bool {
 	return false
 }
 
+// IsLimitConcurrentJobsReached checks if the limit of concurrent jobs by type (backup, check, etc)
+// has been reached
+func (o *Observer) IsLimitConcurrentJobsReached(jobType v1alpha1.JobType, limit int) bool {
```

Comment:
It should be either `IsLimitOfConcurrentJobsReached` or this one:
```suggestion
func (o *Observer) IsConcurrentJobsLimitReached(jobType v1alpha1.JobType, limit int) bool {
```